xds/internal/xdsclient/xds.go

```diff
@@ -561,6 +568,40 @@ func routesProtoToSlice(routes []*v3routepb.Route, logger *grpclog.PrefixLogger,
 	return routesRet, nil
 }
 
+func hashPoliciesProtoToSlice(policies []*v3routepb.RouteAction_HashPolicy) ([]*HashPolicy, error) {
+	var hashPoliciesRet []*HashPolicy
+	for _, p := range policies {
+		// Hash Policies are only applicable for a Ring Hash LB.
+		if !env.RingHashSupport {
+			return nil, errors.New("hash policies are only applicable for ring hash lb policy")
+		}
+		var policy HashPolicy
+		policy.Terminal = p.Terminal
+		switch p.GetPolicySpecifier().(type) {
+		case *v3routepb.RouteAction_HashPolicy_Header_:
+			policy.HashPolicyType = HashPolicyTypeHeader
+			policy.HeaderName = p.GetHeader().HeaderName
```

Comment:
Call `GetHeaderName()`.  Otherwise this will panic if `GetHeader()` returns `nil`.

Reply:
Done. I've actually been wondering about this for a while.